### PR TITLE
switch to newer CITATION.cff schema version

### PR DESCRIPTION
**Pull request by renefritze**
This allows us to list the authors of the software separately from the ones of the paper

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -1,6 +1,27 @@
 # YAML 1.2
 ---
+title: "pyMOR"
 authors: 
+  - family-names: Fritze
+    given-names: "René"
```
Comment (pmli) [inline CITATION.cff:6]: Are quotes necessary here? Should my last name also have them?

Comment (renefritze) [reply CITATION.cff:6]: I think the inconsistencies stem from me starting from a generator website thingy back when, and then sometimes dropping the quotes when editing and sometimes not. 
To make the document valid, I think we only need quotes around numeric-looking values.

Comment (renefritze) [reply CITATION.cff]: The [validator](https://github.com/pymor/pymor/runs/5228483066?check_suite_focus=true) seems happy w/o the quotes.

File: CITATION.cff [hunk 2]
```diff
@@ -1,16 +1,34 @@
...
+title: pyMOR
+authors:
+  - family-names: Fritze
+    given-names: René
+    orcid: https://orcid.org/0000-0002-9548-2238
+  - family-names: Rave
+    given-names: Stephan
+    orcid: https://orcid.org/0000-0003-0439-7212
+  - family-names: Schindler
+    given-names: Felix
```
Comment (ftalbrecht) [inline CITATION.cff:12]:
```suggestion
    given-names: Felix
    orcid: https://orcid.org/0000-0003-1582-7118
```

File: CITATION.cff [hunk 3]
```diff
@@ -1,16 +1,34 @@
...
-    given-names: "René"
-    orcid: "https://orcid.org/0000-0002-9548-2238"
+    given-names: René
+    orcid: https://orcid.org/0000-0002-9548-2238
   - family-names: Rave
     given-names: Stephan
-    orcid: "https://orcid.org/0000-0003-0439-7212"
+    orcid: https://orcid.org/0000-0003-0439-7212
   - family-names: Schindler
     given-names: Felix
```
Comment (ftalbrecht) [inline CITATION.cff:32]:
```suggestion
    given-names: Felix
    orcid: https://orcid.org/0000-0003-1582-7118
```

**Comment by ftalbrecht** [comment]
I suggested to add my orcid id (though my record is not that meaningful). If done correctly, please apply.

**Comment by renefritze** [comment]
The job name in the new workflow file should to be unique across workflows

**Pull request merged**